File: app/src/pfp/java/social/entourage/android/api/PFPApiModule.java
package social.entourage.android.api;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

/**
 * Provides PFP-related API modules
 * Created by Mihai Ionescu on 06/06/2018.
 */
@Module
public class PFPApiModule {
    @Provides
    @Singleton
    public PrivateCircleRequest providesPrivateCircleRequest(final Retrofit restAdapter) {
        return restAdapter.create(PrivateCircleRequest.class);
    }
}

File: app/src/pfp/java/social/entourage/android/api/PFPApiModule.kt
package social.entourage.android.api

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

/**
 * Provides PFP-related API modules
 * Created by Mihai Ionescu on 06/06/2018.
 */
@Module
class PFPApiModule {
    @Provides
    @Singleton
    fun providesPrivateCircleRequest(restAdapter: Retrofit): PrivateCircleRequest {
        return restAdapter.create(PrivateCircleRequest::class.java)
    }
}